### PR TITLE
RET-2730: Add Unassigned to managingOffice

### PR DESCRIPTION
**Pull request by ikirsanov**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RET-2730

### Change description ###
For when we cannot find a managing office to assign to a case, which can occur when a postcode was not provided on the claimant work address or respondent address.

Managing office is added during case submission. If no office is added, as a valid postcode wasn't provided, we must assign it to the default office in the callbacks.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

**Review by JackReevies — APPROVED** [review]

**Review by hsjhita — APPROVED** [review]
@ikirsanov I may have missed a PR but does the Unassigned office exist in the config files?

File: src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/TribunalOfficeServiceTest.java [hunk 1]
```diff
@@ -47,8 +47,8 @@ public class TribunalOfficeServiceTest {
             { TribunalOffice.DUNDEE.getOfficeName(), "DD1 4QB" },
             { TribunalOffice.DUNDEE.getOfficeName(), "DD1 4QB" },
             { TribunalOffice.EDINBURGH.getOfficeName(), "EH3 7HF" },
-            { TribunalOffice.EDINBURGH.getOfficeName(), "EH3 7HF" },
-            { TribunalOffice.EDINBURGH.getOfficeName(), "EH3 7HF" }
+            { "Unassigned", "" },
```
Comment (GauravChawlaMOJ) [inline src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/TribunalOfficeServiceTest.java:50]: Just a minor thing, may be use Unassigned constant here?

**Review by GauravChawlaMOJ — APPROVED** [review]

**Review by fergusoshea — APPROVED** [review]
Not an issue but for Line 24 doesn't "UNASSIGNED_OFFICE.equals(null)" already return false?

**Comment by GauravChawlaMOJ** [comment]
> Not an issue but for Line 24 doesn't "UNASSIGNED_OFFICE.equals(null)" already return false?

@fergusoshea officeName will never be null in UNASSIGNED_OFFICE.equals(officeName) as it is being checked in the condition before. This condition is only checked if office is non-null. (OrElse/ ||)

**Comment by ikirsanov** [comment]
> Not an issue but for Line 24 doesn't "UNASSIGNED_OFFICE.equals(null)" already return false?

If the value will be null, condition will not reach UNASSIGNED_OFFICE.equals(null) I think. If I will leave only UNASSIGNED_OFFICE.equals(value), then in case value == null, this If will be skipped and I dont need this. My situation is If the value is Null or The value is Unassigned then True.

File: src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/TribunalOfficeServiceTest.java (hunk 1 above)
Comment (ikirsanov) [reply src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/TribunalOfficeServiceTest.java]: Added constant to test

**Comment by fergusoshea** [comment]
> If the value will be null, condition will not reach UNASSIGNED_OFFICE.equals(null) I think. If I will leave only UNASSIGNED_OFFICE.equals(value), then in case value == null, this If will be skipped and I dont need this. My situation is If the value is Null or The value is Unassigned then True.

You are right, I misread the second condition. My bad.

**Pull request merged**